### PR TITLE
Make unclosed brace cause a parsing exception

### LINKED ISSUE #242: Un-closed brace does not cause parse error
An open brace with no corresponding close brace does not cause a parse error. For example:
```
42 + [a + 10
```
This is parsed as if there was a closing brace:
```
42 + [a + 10]
```

### PR DESCRIPTION
**Pull request by gmcchessney**
Closes #242

### CODE REVIEW

File: src/NCalc.Core/Parser/LogicalExpressionParser.cs [hunk 1]
```diff
@@ -160,7 +160,7 @@ static LogicalExpressionParser()
 
         // ("[" | "{") identifier ("]" | "}")
         var identifierExpression = openBrace.Or(openCurlyBrace)
-            .SkipAnd(AnyCharBefore(closeBrace.Or(closeCurlyBrace), consumeDelimiter: true))
+            .SkipAnd(AnyCharBefore(closeBrace.Or(closeCurlyBrace), consumeDelimiter: true, failOnEof: true).ElseError("Brace not closed."))
```
Comment (Bykiev) [inline src/NCalc.Core/Parser/LogicalExpressionParser.cs:163]: Thank your for this PR!

Can you please rename "Brace" to "Braces", to match "Parenthesis not closed." error

Comment (gmcchessney) [reply src/NCalc.Core/Parser/LogicalExpressionParser.cs:163]: I don't think that's correct. "Parenthesis" is singular, while "parentheses" is plural.

Comment (gumbarros) [reply src/NCalc.Core/Parser/LogicalExpressionParser.cs:163]: > I don't think that's correct. "Parenthesis" is singular, while "parentheses" is plural.

No English speak here lol, you change this typo if you want

**Review by Bykiev — APPROVED** [review]

**Comment by gumbarros** [comment]
@gmcchessney can you merge _master_ at your branch and add a `ElseError` foreach specific identifier parser?

**Comment by gumbarros** [comment]
@gmcchessney thanks for the contribution

**Pull request merged**